packages/fxa-settings/src/pages/InlineRecoveryKeySetup/container.tsx

```diff
@@ -3,22 +3,145 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import React, { useCallback, useState } from 'react';
-import { Integration } from '../../models';
-import { RouteComponentProps } from '@reach/router';
+import {
+  Integration,
+  useAuthClient,
+  useFtlMsgResolver,
+  useSensitiveDataClient,
+} from '../../models';
+import { RouteComponentProps, useLocation } from '@reach/router';
 import InlineRecoveryKeySetup from '.';
+import { SigninLocationState } from '../Signin/interfaces';
+import { cache } from '../../lib/cache';
+import { generateRecoveryKey } from 'fxa-auth-client/browser';
+import { getHandledError } from '../../lib/error-utils';
+import { CreateRecoveryKeyHandler } from './interfaces';
+import base32Encode from 'base32-encode';
+import { AUTH_DATA_KEY } from '../../lib/sensitive-data-client';
+import { getSyncNavigate } from '../Signin/utils';
+import { hardNavigate } from 'fxa-react/lib/utils';
+import { formatRecoveryKey } from '../../lib/utilities';
 
-export const InlineRecoveryKeySetupContainer = ({
-  integration,
-}: {
-  integration: Integration;
-} & RouteComponentProps) => {
+export const InlineRecoveryKeySetupContainer = (_: RouteComponentProps) => {
   const [currentStep, setCurrentStep] = useState<number>(1);
-  const createRecoveryKeyHandler = useCallback(async () => {
-    // TODO in FXA-10079
-  }, []);
+  const [formattedRecoveryKey, setFormattedRecoveryKey] = useState<string>('');
+  const ftlMsgResolver = useFtlMsgResolver();
+  const authClient = useAuthClient();
+
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state?: SigninLocationState;
+  };
+  const { email, uid, sessionToken, unwrapBKey } = location.state || {};
+
+  const sensitiveDataClient = useSensitiveDataClient();
+  const sensitiveData = sensitiveDataClient.getData(AUTH_DATA_KEY);
+  const { authPW, emailForAuth } =
+    (sensitiveData as unknown as { emailForAuth: string; authPW: string }) ||
```

Comment:
I think we could make some solid improvements with this, and move other values to it. I'll file a follow up and put it in our React Cleanup epic if we don't have something filed already, or leave a note on the React state management spike.